staging/src/k8s.io/client-go/rest/client.go

```diff
@@ -127,6 +132,19 @@ func NewRESTClient(baseURL *url.URL, versionedAPIPath string, config ContentConf
 	}, nil
 }
 
+// RequestTimeoutQuery returns the "timeout" duration sent as part of a request to the API server.
+// When RequestTimeoutQuery() value is <= 0 then it is omitted from API requests.
+func (c *RESTClient) RequestTimeoutQuery() time.Duration {
```

Comment:
Given that there is a proposal in flight to add Context to the rest client in https://github.com/kubernetes/community/pull/1166, I don't think we should be adding direct timeout calculations as a new action on rest client.  Context should be able to cover this use case.